content/A prediction experiment with Machine Learning - Bruno Michetti.md

```diff
@@ -0,0 +1,153 @@
+
+This dataset has 22 columns and 2,938 rows of data. I needed to preprocess the data before training the model. Why did I need that? because when we work in machine learning, we can hardly use the data in the way we get it. First, we need to solve consistency problems, prepare the data for the format and type that the model expects, and remove useless information, etc. If we don’t address those problems, we can’t affirm that the result of the algorithm achieves the main goal. For example, some of the problems could be:
+
+- Missing data.
+- Work with different types of data.
+- Work with different formats of data.
+- Properties of the data with null values.
+- The data format is different from the one that the algorithm expects.
+
+In the next section, I talk about the problems I had to face, and how they can be solved.
+
+### Preprocessing the dataset
+
+The process of applying the necessary transformations to the data in order to prepare the data for the model is called data cleaning. There are several kinds of problems addressed in data cleaning, the ones that I faced are:
+
+- Null values: some of the rows in the dataset had null values, and given that machine learning models are mathematical models, they can’t work with that kind of input.
+- Unnormalized data: some machine learning models don't work well if they have variables that have many different ranges of values.
+- Problems with types: I needed to solve how to pass to a mathematical model, values that weren’t numbered, for example, a categorical variable.
+
+In this section, I talk about some approaches to solve those problems.
+
+#### Null values
+
+When a row has empty spaces, that is null values in the dataset, we can delete that row, or fill those empty spaces with data. I didn’t choose to delete the rows, because I would lose a big part of the dataset, and models don’t work well if they are trained with small datasets or I might be removing relevant information. So I decided to fill the null values. To do this, there are many techniques. For example, choosing a default value, or “simulating a distribution of the data” and selecting a value of that simulation. I chose the second example because using the simulation I’m able to generate numbers similar to the real values. I’ll explain a little bit the method in the next section.
+
+#### Simulate a probability distribution
+
+A random variable is a quantity that is produced by a random process. Besides, a probability distribution is a summary of probabilities for the values of a random variable. Knowing this, we can select a column of the dataset that has null values, and simulate its probability distribution. After this, we can randomly select a value in order to fill the null values affecting our dataset. So for each column in the dataset that has some null values, I executed these steps:
+
+1. Plot a histogram of the column using the [hist function in python](https://matplotlib.org/api/_as_gen/matplotlib.pyplot.hist.html).
```

Comment:
I would say that is a function in matplotlib, rather than python.

Reply:
I'll leave it this way because it is not necessary to talk about matplotlib, given that you don't need to install it.